crates/bevy_ecs/src/system/query.rs

```diff
@@ -321,11 +502,21 @@ impl<'w, 's, Q: WorldQuery, F: WorldQuery> Query<'w, 's, Q, F> {
     /// ```
     #[inline]
     pub fn iter(&self) -> QueryIter<'_, 's, Q::ReadOnly, F::ReadOnly> {
-        // SAFETY: system runs without conflicts with other systems.
-        // same-system queries have runtime borrow checks when they conflict
+        // SAFETY: Aliasing correctness is justified below. (FIXME) `WorldId` being correct is not currently upheld.
+        // `Query::new` does not require the `WorldId` to be correct and even if it did `Query` is constructable
+        // (and fields mutateable) via safe code from anywhere in `bevy_ecs`.
         unsafe {
             self.state.as_readonly().iter_unchecked_manual(
-                self.world,
+                // SAFETY:
+                // -`Query::new` upholds that `QueryWorldBorrows` has adequate access for our `Q` and `F`
+                // `WorldQuery`'s. (FIXME) This is partially wishful thinking as `Query` is constructable (and fields mutateable)
+                // via safe code from anywhere in `bevy_ecs`.
+                // - The `WorldQuery` trait guarantees that `Self::ReadOnly` has a subset of the access that `Self` has,
+                // therefore if we're allowed to access the world with `Q` and `F`(we do, see first point) then we can
+                // access the world using `Q::ReadOnly` and `F::ReadOnly`.
+                // - `ReadOnlyWorldQuery` ensures that `Q::ReadOnly` and `F::ReadOnly` do not access the world mutably
+                // therefore it is fine to access the world with them using a shared reference.
+                self.world.world(),
```

Comment:
The name of these methods results in confusing code when called. Can we rename this to `world_borrow` and `world_borrow_mut`?

Reply:
Yeah I think i agree with this complaint :thinking: probably prefer `.world_ref` and `world_ref_mut` to be consistent with `into_world_ref` :thinking: though perhaps the `world` field should also be renamed to `borrows` or `locks`..